cli/src/command/account.rs

```diff
@@ -926,40 +926,58 @@ async fn print_address(account: &Account, address: &AccountAddress) -> Result<()
     let addresses = account.addresses_with_unspent_outputs().await?;
     let current_time = iota_sdk::utils::unix_timestamp_now().as_secs() as u32;
 
+    let mut output_ids = Vec::new();
+    let mut address_amount = 0;
+    let mut address_nts = Vec::<TokenId>::new();
+    let mut address_nfts = Vec::new();
+    let mut address_aliases = Vec::new();
+    let mut address_foundries = Vec::new();
+
     if let Ok(index) = addresses.binary_search_by_key(&(address.key_index(), address.internal()), |a| {
         (a.key_index(), a.internal())
     }) {
-        let mut address_amount = 0;
-        for output_id in addresses[index].output_ids() {
+        output_ids.extend(addresses[index].output_ids());
+
+        for output_id in &output_ids {
             if let Some(output_data) = account.get_output(output_id).await {
-                // Output might be associated with the address, but can't unlocked by it, so we check that here
+                // Output might be associated with the address, but can't be unlocked by it, so we check that here
+                // TODO: Should we just skip outputs for which this call fails? (that's only possible for
+                // TreasuryOutputs, right?)
                 let (required_address, _) =
                     output_data
                         .output
                         .required_and_unlocked_address(current_time, output_id, None)?;
-                if *address.address().as_ref() == required_address {
+
+                if address.address().as_ref() == &required_address {
+                    if let Some(nts) = output_data.output.native_tokens() {
+                        address_nts.extend(nts.iter().map(|nt| nt.token_id()));
```

Comment:
If two outputs have the same native token, it would create two entries this way, so should probably be a HashSet where you insert or maybe NativeTokensBuilder would be even more interesting, so we can also have the amount?

Reply:
yeah, good point

Reply:
changed it to use the `NativeTokensBuilder`.